tools/disass/ida/disass.py

```diff
@@ -154,7 +238,11 @@ def get_static_successors(inst):
       yield inst.next_ea  # Not recognised as a `noreturn` function.
 
   if inst.is_call():  # Indirect function call, system call.
-    yield inst.next_ea
+    if inst_terminates(inst):
```

Comment:
This looks to be a bug, it should be `inst_terminates_bb`.

I think that a better use of this would be to make the following checks:
 * If the call is a no-return call, then don't `yield inst.next_ea`. It would be nice to have a function like `is_noreturn_call` or something.
 * If the code following the call is not code, according to `is_code`, then don't `yield inst.next_ea`.

The semantics of this instruction are to go and find the natural successors of a particular instruction, not of a block, and so recognizing whether or not a successor instruction is a block terminator (in the sense as it's another block head) does not fit into the semantics of this function. The `analyse_subroutine` function handles the identification and more fine-grained splitting of block heads, using the loop starting at line 306.